### DISCUSSION TITLE
Git Hooks

### DISCUSSION THREAD

**Discussion by MarcelFox** [brazilian-utils/brutils-python#258] [Ideias]
## Contexto

Atualmente o projeto utiliza o hook de _pre-commit_ para formatar o código. Embora a formatação e padronização do código seja necessária e útil no projeto, executar isso a cada _commit_ me parece desnecessário, até atrapalha em certo ponto (para aqueles que gostam de dividir cada commit, deixando explicito o que foi modificado a cada mudança).

## Proposta

Podemos mudar o hook de  _pre-commit_ para um _pre-push hook_. Assim o código de formatação será verificado apenas antes do push _(possívelmente deixando um último commit de linter, sempre que o desenvolvimento for finalizado)_

#### PR: Decidi abrir a discussão para não abrir uma issue direto, como é apenas uma ideia de mudança, acho que vale a discussão.

**Comment by antoniamaia** [comment]
Boa ideia @MarcelFox, vamos abrir uma issue para essa!! Obrigada pela contribuição :)

**Reply by MarcelFox** [reply]
Se quiser posso trabalhar na issue também!

**Reply by camilamaia** [reply]
Criada! https://github.com/brazilian-utils/brutils-python/issues/288. Só dar um okay lá na issue que eu te dou assign. Muito obrigada! Adorei a ideia :)

**Reply by MarcelFox** [reply]
Feito, vale a pena testar o código da branch 288, basta comitar algo no código que não passa no check do _ruff_ e tentar dar o push da branch pra ver se rola a mensagem de erro.

Da forma que ficou, o usuário é informado que um erro ocorreu:
![image](https://github.com/brazilian-utils/brutils-python/assets/4885041/b22e18a1-5b87-4005-800f-15c9e1363f97)
